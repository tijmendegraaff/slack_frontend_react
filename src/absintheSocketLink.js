import * as AbsintheSocket from '@absinthe/socket'
import { createAbsintheSocketLink } from '@absinthe/socket-apollo-link'
import { Socket as PhoenixSocket } from 'phoenix'

// export default createAbsintheSocketLink(AbsintheSocket.create(new PhoenixSocket('ws://localhost:4000/socket')))
const token = localStorage.getItem('token')
export default createAbsintheSocketLink(AbsintheSocket.create(new PhoenixSocket('ws://localhost:4000/socket', {
    reconnect: true,
    params: { token }
})))
