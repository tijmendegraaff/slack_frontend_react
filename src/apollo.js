import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'
import { ApolloLink, concat } from 'apollo-link'
import { InMemoryCache } from 'apollo-cache-inmemory'

const httpLink = new HttpLink({
    uri: `http://${process.env.REACT_APP_SERVER_URL}/api/graphql`
})

const authMiddleware = new ApolloLink((operation, forward) => {
    // add the authorization to the headers
    operation.setContext({
        headers: {
            authorization: `Bearer ${localStorage.getItem('token')}` || null
        }
    })
    return forward(operation)
})

export default new ApolloClient({
    link: concat(authMiddleware, httpLink),
    cache: new InMemoryCache()
})
