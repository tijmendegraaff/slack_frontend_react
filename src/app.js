import React from "react";
import ReactDOM from "react-dom";
import { ApolloProvider } from "react-apollo";
import "normalize.css/normalize.css";

import "./styles/styles.scss";
import client from "./apollo";

const App = (
  <ApolloProvider client={client}>
    <div>hello</div>
  </ApolloProvider>
);

ReactDOM.render(App, document.getElementById("app"));
