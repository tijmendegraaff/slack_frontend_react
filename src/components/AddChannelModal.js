import React from 'react'
import { Modal, Header, Image, Input, Form } from 'semantic-ui-react'

const AddChannelModal = ({
    open, onCloseAddChannelModal, channelName, onChange
}) => (
    <Modal open={open} className="modal-container" onClose={onCloseAddChannelModal}>
        <Modal.Header>Select a Photo</Modal.Header>
        <Modal.Content image>
            <Modal.Description>
                <Header>Default Profile Image</Header>
                <Form.Field>
                    <label>channelName</label>
                    <input
                        placeholder="channel name"
                        name="channelName"
                        onChange={onChange}
                        value={channelName}
                    />
                </Form.Field>
            </Modal.Description>
        </Modal.Content>
    </Modal>
)

export default AddChannelModal
