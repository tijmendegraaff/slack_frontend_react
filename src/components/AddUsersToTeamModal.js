import React from 'react'
import { Modal, Header, Input, Form, Button } from 'semantic-ui-react'

const AddUsersToTeamModal = ({
    open,
    toggleAddUsersToTeamModal,
    addUserEmail,
    addUserEmailError,
    onChange,
    handleAddUsersToTeamSubmit,
    isSubmitting,
    teamId
}) => (
    <Modal open={open} className="modal-container" onClose={toggleAddUsersToTeamModal}>
        <Modal.Header>Add a users to this team!</Modal.Header>
        <Modal.Content>
            <Header>Add users to your team!</Header>
            <Form>
                <Form.Field>
                    <Input
                        placeholder="email"
                        name="addUserEmail"
                        onChange={onChange}
                        value={addUserEmail}
                        fluid
                    />
                </Form.Field>
                <Form.Group widths="equal">
                    <Button disabled={isSubmitting} onClick={handleAddUsersToTeamSubmit} fluid>
                        Add user
                    </Button>
                    <Button disabled={isSubmitting} onClick={toggleAddUsersToTeamModal} fluid>
                        Cancel
                    </Button>
                </Form.Group>
            </Form>
        </Modal.Content>
    </Modal>
)

export default AddUsersToTeamModal
