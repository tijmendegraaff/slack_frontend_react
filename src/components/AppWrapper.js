import React from 'react'
import PropTypes from 'prop-types'

const AppWrapper = props => <div className="app-layout">{props.children}</div>

AppWrapper.propTypes = {
    children: PropTypes.any.isRequired
}

export default AppWrapper
