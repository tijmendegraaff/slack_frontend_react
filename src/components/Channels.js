import React from 'react'
import PropTypes from 'prop-types'

// eslint-disable-next-line
const channel = ({ id, name }) => (
    <li key={`channel-${id}`} className="channel-list-item">
        # {name}
    </li>
)

const ActiveUserSpan = () => <span className="blue">●</span>

// eslint-disable-next-line
const user = ({ id, name, active }) => (
    <li key={`user-${id}`} className="channel-list-item">
        {!active ? <ActiveUserSpan /> : '○'} {name}
    </li>
)

const Channels = ({
    teamName, username, channels, users
}) => (
    <div className="channels">
        <h1 className="channel-title">{teamName}</h1>
        <h1 className="channel-username">{username}</h1>
        <ul className="channel-list">
            <li className="channel-list-header">Channels</li>
            {channels.map(channel)}
        </ul>
        <ul className="channel-list">
            <li className="channel-list-header">Users</li>
            {users.map(user)}
        </ul>
    </div>
)

Channels.propTypes = {
    teamName: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    channels: PropTypes.array.isRequired,
    users: PropTypes.array.isRequired
}

export default Channels
