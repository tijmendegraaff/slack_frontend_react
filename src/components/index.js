import AppWrapper from './AppWrapper'
import Channels from './Channels'
import HeaderWrapper from './HeaderWrapper'
import Teams from './Teams'
import SendMessage from './SendMessage'
import AddChannelModal from './AddChannelModal'

export { AppWrapper, Channels, HeaderWrapper, Teams, SendMessage, AddChannelModal }
