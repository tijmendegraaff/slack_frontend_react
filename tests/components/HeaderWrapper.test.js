import React from 'react'
import { shallow } from 'enzyme'
import { HeaderWrapper } from '../../src/components'

test('should render the HeaderWrapper correctly', () => {
    const wrapper = shallow(<HeaderWrapper channelName="general" />)
    expect(wrapper).toMatchSnapshot()
})
