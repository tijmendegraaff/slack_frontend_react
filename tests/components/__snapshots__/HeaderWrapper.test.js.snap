// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the HeaderWrapper correctly 1`] = `
<div
  className="header"
>
  <Header
    textAlign="center"
  >
    #
    general
  </Header>
</div>
`;
